packages/dht/src/connection/WebSocket/WebSocketConnector.ts

```diff
@@ -102,39 +101,39 @@ export class WebSocketConnector implements IWebSocketConnectorService {
                     this.attachHandshaker(connection)
                 }
             })
-            await this.webSocketServer.start(this.webSocketPort!, this.webSocketHost)
+            const port = await this.webSocketServer.start(this.portRange!, this.host)
+            this.selectedPort = port
+            this.connectivityChecker = new ConnectivityChecker(this.selectedPort!)
         }
     }
 
     public async checkConnectivity(reattempt = 0): Promise<ConnectivityResponse> {
+        const noServerConnectivityResponse: ConnectivityResponse = {
+            openInternet: false,
+            ip: '127.0.0.1',
+            natType: NatType.UNKNOWN
+        }
+        if (this.destroyed) {
+            return noServerConnectivityResponse
+        }
         try {
-            const noServerConnectivityResponse: ConnectivityResponse = {
-                openInternet: false,
-                ip: '127.0.0.1',
-                natType: NatType.UNKNOWN
-            }
-
             if (!this.webSocketServer) {
                 // If no websocket server, return openInternet: false
                 return noServerConnectivityResponse
             } else {
                 if (!this.entrypoints || this.entrypoints.length < 1) {
```

Comment:
It would maybe make sense to normalize the entrypoints array so that we can treat empty array and missing array equally. The normalization should be done already in `DhtNodeConfig`. Maybe a separate PR about this?